backend/tests/conftest.py

```diff
@@ -34,8 +39,14 @@ def logged_in_client(test_client) -> str:
 
 
 @pytest.fixture()
-def test_project_id(logged_in_client, test_project_name):
+def created_project_id(logged_in_client, test_project_name):
     data = {"name": test_project_name}
     response = logged_in_client.post(f"{API_VERSION_PREFIX}/projects", json=data)
     json_result = response.json()
     return json_result.get("id")
+
+
+@pytest.fixture()
+def test_project_id(logged_in_client, created_project_id):
+    yield created_project_id
+    logged_in_client.delete(f"{API_VERSION_PREFIX}/projects/{created_project_id}")
```

Comment:
Good ; you don't need to use the API to create/delete your fixtures. It makes testing your API more difficult anyway

Reply:
> Good ; you don't need to use the API to create/delete your fixtures. It makes testing your API more difficult anyway

I don't get it here. If I don't use the API to create/delete project, How I can create/delete the project?

Reply:
Those tests are integration tests. They test endpoints behavior.
If you want to test endpoints independently (that's what we do we our per-endpoint tests), you may not want to depend on another endpoint behaving correctly. Otherwise, if that dependency endpoint fails, then all your tests fail and you have to fix that first because without it fixed, your tests are useless.
To avoid this, you can use the DB to create your fixtures (insert and delete data) ; assuming you already, independently test all endpoints (and eventually end-to-end test)
But all approaches are valid; just wanted to make sure you knew your options.